### ISSUE TITLE
QDialog triggered from QPopupEdit behaves unexpectedly on click

### ISSUE THREAD

**Issue by elnicko** [quasarframework/quasar#10449]
**Describe the bug**
I'd like to use custom inputs in a QPopupEdit. Some of them come with a button in the append slot that opens an embedded QDialog. However, the opened QDialog seem to not be really usable, as when clicking inside it, the QPopupEdit disappears and so does any opened QDialog. The problem seems to be similar to [issue #6737](https://github.com/quasarframework/quasar/issues/6737), as the click happens outside the QPopupEdit.

Any suggestion on how to work around it or fix it really appreciated.

**Codepen/jsFiddle/Codesandbox (required)**
https://jsfiddle.net/d0m4x18r/2/

**To Reproduce**
1. Open the linked jsFiddle
2. Click on 'Click me'
3. Click on the q-btn in the '#append' slot of the q-input. A dialog with some text appears.
4. Now try to select the text
5. The dialog disappears unexpectedly

**Expected behavior**
The dialog does not disappear.

**Screenshots**
Not needed, see the codepen

**Platform (please complete the following information):**
Quasar Version: ^2.0.1
@quasar/app Version: ^3.0.1

Quasar mode:
  - [x] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

Tested on:
  - [x] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

OS: independent
Node: >= 14.17.3
NPM: >= 7.20.1
Yarn: >= 1.21.1
Browsers: Chrome/FF
iOS:
Android:
Electron:

**Additional context**
Not needed

**Comment by xwyzsn** [comment]
you may move the dialog section **outside of** QPopupEdit 
```html
  <div class="cursor-pointer">
      {{ label }}
      <q-popup-edit v-model="label" auto-save v-slot="scope">
        <q-input v-model="scope.value" dense autofocus counter @keyup.enter="scope.set">
          <template #append>
            <q-btn label="dialog" @click="dialogModel = true"/>
          </template>
        </q-input>
      </q-popup-edit>
      <q-dialog v-model="dialogModel">
          <q-card>
            <q-card-section>
              <div class="text-h6">Alert</div>
              <div class="text-h6">Alert</div>
              <div class="text-h6">Alert</div>
              <div class="text-h6">Alert</div>
              <div class="text-h6">Alert</div>
              <div class="text-h6">Alert</div>
              <div class="text-h6">Alert</div>
              <div class="text-h6">Alert</div>
              <div class="text-h6">Alert</div>
            </q-card-section>
            </q-card>
        </q-dialog>
    </div>
```
it should be fixed

**Comment by elnicko** [comment]
Thanks for the out-of-the-box suggestion. I'm afraid however that moving the dialog outside is not an option. 

The inputs are containers that are bundled separately and cannot be changed. They contain the QDialog already. They need to work the same regardless of whether inside a QPopupEdit or any other parent/context.

The example is a single component just to illustrate the issue in the most straight-to-the-point way.

**Comment by elnicko** [comment]
Any ideas?

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v2.1.3
